added code format to see type of code scanned

android-imdad opened the pull request:
## Summary

Added codeFormat to see what kind of code is being scanned, useful when there is code specific validation that needs to be done.

## How did you test this change?

Tested example apps on both IOS and Android, no visual change for screenshot.

DavidBertet commented on ios/ReactNativeCameraKit/SimulatorCamera.swift:
```diff
@@ -164,8 +164,8 @@ class SimulatorCamera: CameraProtocol {
     func isBarcodeScannerEnabled(_ isEnabled: Bool,
-                                 supportedBarcodeType: [AVMetadataObject.ObjectType],
-                                 onBarcodeRead: ((_ barcode: String) -> Void)?) {}
+                                 supportedBarcodeTypes supportedBarcodeType: [AVMetadataObject.ObjectType],
```
Let's remove old `supportedBarcodeType`

Shouldn't we update this type as well, so it takes `CodeFormat` as input and output?

```suggestion
                                 supportedBarcodeTypes: [CodeFormat],
```

DavidBertet commented on android/src/main/java/com/rncamerakit/CodeFormat.kt:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    companion object {
+        fun fromBarcodeType(barcodeType: Int): CodeFormat =
```
We can add this annotation to enforce what kind of `Int` we are receiving here: `@Barcode.BarcodeFormat`

```suggestion
        fun fromBarcodeType(@Barcode.BarcodeFormat barcodeType: Int): CodeFormat =
```

It might work as well on the return type of `toBarcodeType`

DavidBertet commented on src/Camera.d.ts:
```diff
@@ -4,6 +4,7 @@ import { Orientation } from './index';
 export type OnReadCodeData = {
   nativeEvent: {
     codeStringValue: string;
+    codeFormat: string;
```
The expected strings are known, could you use the enum notation instead?

// types.ts
export type CodeFormat = 'code-128' | 'code-39' | ...;

// Camera.d.ts
codeFormat: CodeFormat;

android-imdad replied:
Added this on commit 4af9bac

DavidBertet reviewed (commented):
I was planning to do it next week, thanks @android-imdad!

A few feedbacks, but it's looking great. Can you also check the overall syntax, spaces, blank lines, file header, to be consistent with other places?

android-imdad commented:
Hi, yes will check these comments in a few days

android-imdad commented:
@DavidBertet I fixed the indentation, accepted your changes and added types. Please review and let me know if there are any concerns.

DavidBertet commented:
> @DavidBertet I fixed the indentation, accepted your changes and added types. Please review and let me know if there are any concerns.

Thanks @android-imdad. Have you tried it? I doubt the iOS part compiles at all.

When I said to replace `AVMetadataObject.ObjectType` by `CodeFormat`, there is more than just changing a type. It has to be reflected all along the chain. On both camera & simulator, as well as in `CameraProtocol`

android-imdad commented:
> > @DavidBertet I fixed the indentation, accepted your changes and added types. Please review and let me know if there are any concerns.
> 
> Thanks @android-imdad. Have you tried it? I doubt the iOS part compiles at all.
> 
> When I said to replace `AVMetadataObject.ObjectType` by `CodeFormat`, there is more than just changing a type. It has to be reflected all along the chain. On both camera & simulator, as well as in `CameraProtocol`

My bad, nope didn't try it after the change. Will test it and do a commit with the fixes.

android-imdad commented:
@DavidBertet Please check now, I made the changes and tested in both IOS and Android, seems to be working as expected.

DavidBertet commented on ios/ReactNativeCameraKit/RealCamera.swift:
```diff
@@ -484,8 +485,9 @@ class RealCamera: NSObject, CameraProtocol, AVCaptureMetadataOutputObjectsDelega
-            metadataOutput.metadataObjectTypes = filteredTypes
+            let filteredTypes = supportedBarcodeType.filter { type in availableTypes.contains(type.toAVMetadataObjectType()) }
+
+            metadataOutput.metadataObjectTypes = filteredTypes.map { $0.toAVMetadataObjectType() }
```
We are doing `toAVMetadataObjectType` twice for the same objects. Something like that should do the same
```
let filteredTypes = supportedBarcodeType
  .map { $0.toAVMetadataObjectType() }
  .filter { availableTypes.contains($0) }

metadataOutput.metadataObjectTypes = filteredTypes
```

DavidBertet commented on ios/ReactNativeCameraKit/CameraView.swift:
```diff
@@ -184,14 +186,16 @@ class CameraView: UIView {
 
         // Scanner
         if changedProps.contains("scanBarcode") || changedProps.contains("onReadCode") {
+            let supportedFormats = supportedBarcodeType.map { CodeFormat.fromAVMetadataObjectType($0) }
```
We can use `CodeFormat` for `supportedBarcodeType` line 23, so no need to map it here

android-imdad replied:
Oh yes I should have just done that :D

android-imdad replied:
@DavidBertet  check now

DavidBertet reviewed (approved):
I couldn't test for now, 2 minor feedbacks but the code looks right. 

Thanks @android-imdad for your contribution!

scarlac commented on src/Camera.d.ts:
```diff
@@ -1,9 +1,10 @@
-import { CameraApi, FlashMode, FocusMode, ZoomMode, TorchMode, CameraType } from './types';
+import { CameraApi, FlashMode, FocusMode, ZoomMode, TorchMode, CameraType,CodeFormat } from './types';
```
```suggestion
import { CameraApi, FlashMode, FocusMode, ZoomMode, TorchMode, CameraType, CodeFormat } from './types';
```

scarlac commented:
Thanks @android-imdad and @DavidBertet !
